### DISCUSSION TITLE
Validate JWT token with JWKs

### DISCUSSION THREAD

**Discussion by apchee** [pocoproject/poco#3288] [Q&A]
I know I can use Poco/JWT generates, encodes and validates a JWT token, but how it can validate a JWT token with JWKs? Is there any example?

**Comment by monhi** [comment]
I have the same question. 
A sample code will help.

**Comment by CharlesMcBaker** [comment]
Also interested

**Comment by Spaky** [comment]
If you have the JWKS, do can do this way:
* extract public key from JWKS
* create a "Poco::JWT::Signer"
* set pub key by "setECKey" or "setRSAKey" for example depending on kty to signer
* add algorithm based on alg by "addAlgorithm" to signer
* call signer verify function
* compare issuser
* check not before date
* check expiration date
* check additional claims if you need
